docs/en/how_to_guides/finetune_with_a_custom_dataset.md

```diff
@@ -0,0 +1,357 @@
+
+For hyper-parameters used in fine-tuning training, you can refer to the configuration file used when pre-training on the ImageNet-1k dataset in ./configs. Note that for fine-tuning, <font color=DarkRed>the hyper-parameter `pretrained` should be set to be `True` </font>to load the pre-training weight, <font color=DarkRed> `num_classes` should be set to be the number of labels </font>of the custom dataset (e.g. 100 for the Aircraft dataset here), moreover, don't forget to <font color=DarkRed>reduce batch_size and epoch_size </font>based on the size of the custom dataset. In addition, since the pre-trained weight already contains a lot of information for identifying images, in order not to destroy this information too much, it is also necessary to<font color=DarkRed> reduce the learning rate `lr` </font>, and it is also recommended to start training and adjust from at most one-tenth of the pre-trained learning rate or 0.0001. These parameters can be modified in the configuration file or added in the shell command as shown below. The training results can be viewed in the file ./ckpt/results.txt.
+
+```bash
+python finetune.py --config=./configs/mobilenetv3/mobilnet_v3_small_ascend.yaml --data_dir=./aircraft/data --pretrained=True
+```
+
+When fine-tuning mobilenet v3-small based on Aircraft dataset, this tutorial mainly made the following changes to the hyper-parameters:
+
+| Hyper-parameter | Before     | After                |
+| --------------- | ---------- | -------------------- |
+| dataset         | "imagenet" | ""                   |
+| batch_size      | 75         | 8                    |
+| image_resize    | 224        | 600                  |
+| auto_augment    | -          | "randaug-m7-mstd0.5" |
+| num_classes     | 1000       | 100                  |
+| pretrained      | False      | True                 |
+| epoch_size      | 470        | 50                   |
+| lr              | 0.77       | 0.002                |
+
+### Fine-tuning All the Parameters
+
+Since the progress of this type of fine-tuning is the same as training from scratch, simply **start the training by running finetune.py ** and adjust the parameters as training from scratch.
+
+### Freeze Feature Network
+
+#### Linear Probe
+
+We prevent parameters from updating by setting `requires_grad=False` for all parameters except those in the full connection layer. In finetune.py, add the following code after `create_model` :
+
+```python
+from mindcv.models.registry import _model_pretrained_cfgs
+
+# ...create_model()
+
+# number of parameters to be updated
+num_params = 2
+
+# read names of parameters in FC layer
+classifier_names = [_model_pretrained_cfgs[args.model]["classifier"] + ".weight",
+                    _model_pretrained_cfgs[args.model]["classifier"] + ".bias"]
+
+# prevent parameters in network(except the classifier) from updating
+for param in network.trainable_params():
+    if param.name not in classifier_names:
+        param.requires_grad = False
+```
```

Comment:
It should be better to wrap these code of freezing backbone into a function, allowing easy integration for other datasets/tasks.

```python
def freeze_backbone(net, cfg):
   ...
   return net

```